src/Features/Core/Portable/UseObjectInitializer/AbstractUseObjectInitializerDiagnosticAnalyzer.cs

```diff
@@ -133,281 +142,77 @@ public DiagnosticAnalyzerCategory GetAnalyzerCategory()
         {
             return DiagnosticAnalyzerCategory.SemanticDocumentAnalysis;
         }
-    }
-
-    internal struct Match<TAssignmentStatementSyntax, TMemberAccessExpressionSyntax, TExpressionSyntax>
-        where TExpressionSyntax : SyntaxNode
-        where TMemberAccessExpressionSyntax : TExpressionSyntax
-        where TAssignmentStatementSyntax : SyntaxNode
-    {
-        public readonly TAssignmentStatementSyntax Statement;
-        public readonly TMemberAccessExpressionSyntax MemberAccessExpression;
-        public readonly TExpressionSyntax Initializer;
-
-        public Match(
-            TAssignmentStatementSyntax statement,
-            TMemberAccessExpressionSyntax memberAccessExpression,
-            TExpressionSyntax initializer)
-        {
-            Statement = statement;
-            MemberAccessExpression = memberAccessExpression;
-            Initializer = initializer;
-        }
-    }
-
-    internal struct Analyzer<
-            TExpressionSyntax,
-            TStatementSyntax, 
-            TObjectCreationExpressionSyntax, 
-            TMemberAccessExpressionSyntax,
-            TAssignmentStatementSyntax,
-            TVariableDeclaratorSyntax>
-        where TExpressionSyntax : SyntaxNode
-        where TStatementSyntax : SyntaxNode
-        where TObjectCreationExpressionSyntax : TExpressionSyntax
-        where TMemberAccessExpressionSyntax : TExpressionSyntax
-        where TAssignmentStatementSyntax : TStatementSyntax
-        where TVariableDeclaratorSyntax : SyntaxNode
-    {
-        private readonly ISyntaxFactsService _syntaxFacts;
-        private readonly TObjectCreationExpressionSyntax _objectCreationExpression;
-
-        private TStatementSyntax _containingStatement;
-        private SyntaxNodeOrToken _valuePattern;
-
-        public Analyzer(
-            ISyntaxFactsService syntaxFacts, 
-            TObjectCreationExpressionSyntax objectCreationExpression) : this()
-        {
-            _syntaxFacts = syntaxFacts;
-            _objectCreationExpression = objectCreationExpression;
-        }
 
-        internal ImmutableArray<Match<TAssignmentStatementSyntax, TMemberAccessExpressionSyntax, TExpressionSyntax>> Analyze()
+        public Task FixAllAsync(
+            Document document, ImmutableArray<Diagnostic> diagnostics,
+            SyntaxEditor editor, CancellationToken cancellationToken)
```

Comment:
this is the part that really got added.  A new way of doing FixAll that allows us to do it as tree transformations instead of text edits, and which properly handles all sorts of nasty nested cases.